src/main/java/net/fabricmc/loom/task/MigrateMappingsTask.java

```diff
@@ -107,42 +112,61 @@ public void doTask() throws Throwable {
 
 		try {
 			TinyTree currentMappings = mappingsProvider.getMappings();
-			TinyTree targetMappings = getMappings(project, targetMappingsName, useLocalMappings);
-			migrateMappings(project, extension.getMinecraftMappedProvider(), inputDirPath, outputDirPath, currentMappings, targetMappings, extension);
+			TinyTree targetMappings = getMappings(mappings);
+			migrateMappings(project, extension.getMinecraftMappedProvider(), inputDirPath, outputDirPath, currentMappings, targetMappings);
 			project.getLogger().lifecycle(":remapped project written to " + outputDirPath.toAbsolutePath());
 		} catch (IOException e) {
-			throw new IllegalArgumentException("Could not find mappings for version " + officialMappingsVersion, e);
+			throw new IllegalArgumentException("Error while loading mappings", e);
 		}
 	}
 
-	private TinyTree getMappings(Project project, String mappingsVersionOrPath, boolean useLocalMappings) throws IOException {
-		Path migrateMappingsDir = Files.createTempDirectory("migrate");
-		Path localMappingsOfVersion = migrateMappingsDir.resolve(mappingsVersionOrPath + ".tiny");
-
-		if (!Files.exists(localMappingsOfVersion)) {
-			if (useLocalMappings) {
-				Files.copy(Paths.get(mappingsVersionOrPath), localMappingsOfVersion);
-			} else {
-				String versionRelativePath = UrlEscapers.urlFragmentEscaper().escape(mappingsVersionOrPath);
-				String artifactUrl = "https://maven.fabricmc.net/net/fabricmc/yarn/" + versionRelativePath + "/yarn-" + versionRelativePath + ".jar";
-				File mappingsJar = File.createTempFile("migrateMappingsJar", ".jar");
-				project.getLogger().lifecycle(":downloading new mappings from " + artifactUrl);
-				FileUtils.copyURLToFile(new URL(artifactUrl), mappingsJar);
-
-				try (FileSystem jar = FileSystems.newFileSystem(mappingsJar.toPath(), null)) {
-					if (!Files.exists(migrateMappingsDir)) Files.createDirectory(migrateMappingsDir);
-					MappingsProvider.extractMappings(jar, localMappingsOfVersion);
-				}
+	private File loadMappings() {
+		Project project = getProject();
+
+		if (mappings == null || mappings.isEmpty()) {
+			throw new IllegalArgumentException("No mappings were specified. Use --mappings=\"\" to specify target mappings");
+		}
+
+		Set<File> files;
+
+		try {
+			files = project.getConfigurations().detachedConfiguration(project.getDependencies().create(mappings)).resolve();
+		} catch (IllegalDependencyNotation ignored) {
+			project.getLogger().info("Could not locate mappings, presuming Yarn V2");
+
+			try {
+				files = project.getConfigurations().detachedConfiguration(project.getDependencies().module(ImmutableMap.of("group", "net.fabricmc", "name", "yarn", "version", mappings, "classifier", "v2"))).resolve();
+			} catch (GradleException ignored2) {
```

Comment:
```suggestion
			} catch (IllegalDependencyNotation ignored2) {
```

Reply:
No? This may throw an exception saying the dependency couldn't be resolved, not that it is invalid notation.

Reply:
What about the previous try/catch then?

Reply:
line 134

Reply:
That catches if the notation was incorrect, not if any dependency wasn't resolved.

Reply:
Alright then